driver-core/src/main/com/mongodb/client/model/AggregationLevel.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+/**
+ * The aggregation level
+ *
+ * Where an aggregation command should take place.
+ *
+ * @since 3.10
+ */
+public enum AggregationLevel {
```

Comment:
For 3.x, this has to be public and in a non-internal package since AggregateOperation depends on it, but although it's a bit weird let's deprecate it since we won't want it public in 4.0.

Reply:
Done and merged